src/js/angular/resources/controllers.js

```diff
@@ -99,12 +128,114 @@ resourcesCtrl.controller('ResourcesCtrl', ['$scope', '$timeout', 'MonitoringRest
                 $scope.loader = false;
             });
         };
+        let queryMonitorData;
+
+        const getQueryMonitor = function() {
+            const activeRepository = $scope.getActiveRepository();
+            if ($scope.error) {
+                return;
+            } else if (!activeRepository) {
+                queryMonitorData = $timeout(getQueryMonitor, POLLING_INTERVAL);
+                return;
+            }
+            $q.all([getPerformanceMonitorData(), getActiveQueryMonitor()])
+                .then((response) => {
+                    const [performanceData, activeQueryData] = response;
+                    const timestamp = new Date();
+
+                    if (performanceData) {
+                        Object.values($scope.performanceMonitorData).forEach((chart) => {
+                            chart.addData(timestamp, performanceData);
+                        });
+                    }
+
+                    if (activeQueryData) {
+                        $scope.queriesChart.addData(timestamp, {activeQueryData, performanceData});
+                    }
+
+                    if ($scope.firstLoad) {
+                        $scope.firstLoad = false;
+
+                        const timer = $timeout(function () {
+                            $scope.loader = false;
+                        }, 500);
+
+                        $scope.$on('$destroy', function () {
+                            $timeout.cancel(timer);
+                        });
+                    }
+
+                    queryMonitorData = $timeout(getQueryMonitor, POLLING_INTERVAL);
+                });
+        };
+        const getPerformanceMonitorData = function () {
+            const activeRepository = $scope.getActiveRepository();
+            return MonitoringRestService.monitorQueryTransactionStatistics(activeRepository).then(function (response) {
+                return response.data;
+            }).catch(function (data) {
+                $scope.error = getError(data);
```

Comment:
This is quite nasty way for error handling. Right now all request errors end up in this global error variable and they potentially can override each other. Can you centralize the logging by just handling it up the promise chain in a single place?

Reply:
Will refactor error handling in a separate issue.